### PR TITLE
Maximize usage of inheritance in browser tests #6462

### LINKED ISSUE #6462: Maximize usage of inheritance in browser tests
All browser tests inherits `BaseUiTestCase` but its usage of inheritance is extremely limited. We can have things like making `Browser` and `DataBundle` `protected` variables of `BaseUiTestCase` and this would suppress all the repetitions in all the browser tests.

While not every single repetition is a candidate for abstraction (we have seen how wrong abstraction can kill), many stuffs are clear abstraction candidate, such as the two aforementioned items.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6462 

**Outline of Solution**

Abstractions made for:
- `Browser` and `DataBundle` variables as explained in the issue thread.
- `BeforeClass` operations: no more do we need to repeat `printTestClassHeader`, `browser = BrowserPool.getBrowser()`, and both are correct abstractions because they are common for every single browser tests.
- `AfterClass` operations: ditto, but the operations are `printTestClassFooter` and `BrowserPool.release(browser)`. Again, correct abstractions.
- `prepareTestData()`: all browser tests must not forget to prepare the data needed for their tests. This being an `abstract` method will force all new browser tests to be compliant.
- `executeScript()`: preventing the repetitive usage of `(JavascriptExecutor) browser.driver` everywhere.

The last two commits are fixes for non-standard practices.

### CODE REVIEW

**Comment by junhaoyap** [comment]
@wkurniawan07 same here, when I wake up

**Comment by junhaoyap** [comment]
Finally got here after looking through various things and CNY eve stuff, thanks for waiting patiently

File: src/test/java/teammates/test/cases/browsertests/BaseUiTestCase.java [hunk 1]
```diff
@@ -31,6 +38,25 @@ public static void checkAndEnableGodMode() {
...
     }
+    
+    @BeforeClass
+    public void baseClassSetup() throws Exception {
+        printTestClassHeader();
+        prepareTestData();
+        browser = BrowserPool.getBrowser();
+    }
+    
+    protected abstract void prepareTestData() throws Exception;
```
Comment (junhaoyap) [inline src/test/java/teammates/test/cases/browsertests/BaseUiTestCase.java]: I really like this

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 2]
```diff
@@ -144,8 +144,6 @@ private void testEditSessionAction() throws Exception {
                 editedSession.getCourseId(), editedSession.getFeedbackSessionName());
         editedSession.setInstructions(new Text("<p>" + editedSession.getInstructionsString() + "</p>"));
         assertEquals(editedSession.toString(), savedSession.toString());
-        assertEquals("overflow-auto alert alert-success statusMessage",
```
Comment (junhaoyap) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java]: Is this useless and that is why it is removed?

File: src/test/java/teammates/test/cases/browsertests/AdminEmailLogPageUiTest.java [hunk 3]
```diff
@@ -5,27 +5,21 @@
...
-    private static Browser browser;
     private static AdminEmailLogPage emailLogPage;
     
-    @BeforeClass
-    public void classSetup() {
-        printTestClassHeader();
-        browser = BrowserPool.getBrowser();
+    @Override
+    protected void prepareTestData() {
+        // no test data used in this test
```
Comment (junhaoyap) [inline src/test/java/teammates/test/cases/browsertests/AdminEmailLogPageUiTest.java]: I like this comment

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java (hunk 2 above)
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java]: Not entirely useless, but not value-adding anything imo. Why would anyone test the _HTML class_ of something?

File: src/test/java/teammates/test/cases/browsertests/AllAccessControlUiTests.java [hunk 4]
```diff
@@ -115,13 +98,13 @@ public void testUserNotRegistered() {
 
     @Test
     public void testStudentAccessToAdminPages() {
-        loginStudent(studentUsername, studentPassword);
+        loginStudent(TestProperties.TEST_STUDENT1_ACCOUNT, TestProperties.TEST_STUDENT2_ACCOUNT);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/browsertests/AllAccessControlUiTests.java]: 2nd parameter is wrong

File: src/test/java/teammates/test/cases/browsertests/AllAccessControlUiTests.java [hunk 5]
```diff
@@ -115,13 +98,13 @@ public void testUserNotRegistered() {
...
         verifyCannotAccessAdminPages();
     }
 
     @Test
     public void testStudentHome() {
-        loginStudent(studentUsername, studentPassword);
+        loginStudent(TestProperties.TEST_STUDENT1_ACCOUNT, TestProperties.TEST_STUDENT2_ACCOUNT);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/browsertests/AllAccessControlUiTests.java]: 2nd parameter is wrong here too.

**Comment by wkurniawan07** [comment]
Will do now. Apologies for this!

**Pull request merged**